src/main/webapp/3Dscript.js

```diff
@@ -57,143 +88,261 @@ function fetchData() {
   fetch('/getrun?id=' + id + '&dataType=' + type)
       .then((response) => response.json())
       .then((data) => pose = data)
-      .then(() => addMap());
+      .then(() => {
+        const [lat, lng, alt] = findMedians();
+        const latLng = {latitude, longitude};
+        const origin = new Point(lat, lng, alt);
+        addMap(latLng);
+        const [xAxis, yAxis, zAxis] = createInstances();
+        const orientation = {x: xAxis, y: yAxis, z: zAxis};
+        plotTrajectory(origin);
+        plotOrientation(orientation, origin);
+      });
+}
+
+/**
+ * Finds the median value of the latitude, longitude,
+ * and altitude. The median is used as a reference point
+ * for the East-North-Up (ENU) conversion and as the center
+ * point for the 2D map.
+ * @return {array} An array containing median lat, lng, and alt
+ *   coordinates.
+ */
+function findMedians() {
+  const arrays = {lat: [], lng: [], alt: []};
+  for (const point of pose) {
+    arrays.lat.push(point.lat);
+    arrays.lng.push(point.lng);
+    arrays.alt.push(point.alt);
+  }
+  const sorted = {
+    lat: [...arrays.lat].sort((a, b) => a - b),
+    lng: [...arrays.lng].sort((a, b) => a - b),
+    alt: [...arrays.alt].sort((a, b) => a - b)};
+  const mid = {
+    lat: sorted.lat[Math.floor(sorted.lat.length / 2)],
+    lng: sorted.lng[Math.floor(sorted.lng.length / 2)],
+    alt: sorted.alt[Math.floor(sorted.alt.length / 2)]};
+  return [mid.lat, mid.lng, mid.alt];
 }
 
 /**
  * Creates the 2D terrain and points a light at it.
  * It also calls two other functions so it can wait on the fetch
  * request and significantly cut down rendering time with
  * geometry instancing.
+ * @param {Object} center An LLA coordinate used as the center
+ *     of the 2D map.
 */
-function addMap() {
+function addMap(center) {
   // Creates the 2D map.
   const loader = new THREE.TextureLoader();
-  let material = new THREE.MeshLambertMaterial({
+  const material = new THREE.MeshLambertMaterial({
     map: loader.load(
         'https://maps.googleapis.com/maps/api/staticmap' +
-      '?format=png&center=' + pose[0].lat + ',' + pose[0].lng +
+      '?format=png&center=' + center.latitude + ',' + center.longitude +
       '&zoom=18&size=500x500&key=' + apiKey),
   });
-  let geometry = new THREE.PlaneGeometry(50, 50);
+  const geometry = new THREE.PlaneGeometry(200, 200);
   const map = new THREE.Mesh(geometry, material);
-  map.position.set(0, -4, 0);
-  map.rotation.x = -Math.PI / 2;
+  map.position.set(0, 0, -4);
```

Comment:
maybe put magic numbers in variables?

Reply:
Could I maybe put a comment explaining? Adding 3 variables for each axis seems kind of redundant.